go/pkg/array/reduce_test.go

```
@@ -0,0 +1,546 @@
+	t.Run("sums integers", func(t *testing.T) {
+		numbers := []int{1, 2, 3, 4, 5}
+		sum := Reduce(numbers, func(acc, val int) int {
+			return acc + val
+		}, 0)
+
+		require.Equal(t, 15, sum)
+	})
+
+	t.Run("finds maximum value", func(t *testing.T) {
+		values := []float64{3.14, 2.71, 1.41, 4.67, 2.23}
+		max := Reduce(values, func(acc, val float64) float64 {
```

golangci-lint (2.2.2)

variable max has same name as predeclared identifier

(predeclared)